Framework/DataObjects/src/EventList.cpp

```diff
@@ -3106,6 +3147,56 @@ DateAndTime EventList::getPulseTimeMax() const {
   return tMax;
 }
 
+void EventList::getPulseTimeMinMax(Mantid::Kernel::DateAndTime &tMin,
+                                   Mantid::Kernel::DateAndTime &tMax) const {
+  // set up as the minimum available date time.
+  tMax = DateAndTime::minimum();
+  tMin = DateAndTime::maximum();
+
+  // no events is a soft error
+  if (this->empty())
+    return;
+
+  // when events are ordered by pulse time just need the first/last values
+  if (this->order == PULSETIME_SORT) {
+    switch (eventType) {
+    case TOF:
+      tMin = this->events.begin()->pulseTime();
+      tMax = this->events.rbegin()->pulseTime();
```

Comment:
I've just copied the expression used in getPulseTimeMin getPulseTimeMax. Following "minimal perturbations to framework" principle, let's live it as it is so would agree it may be better to change it everywhere in the EventList.